src/components/common/CountDownTimer.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div />
+</template>
+
+<script>
+import VueTimers from 'vue-timers/mixin';
+
+export default {
+  name: 'CountDownTimer',
+
+  props: {
+    timeout: {
+      type: Number,
+      default: 30000,
+    },
+    isLocked: {
+      type: Boolean,
+      default: false,
+    },
+    counter: {
+      type: Number,
+      default: 0,
+    },
+  },
+
+  computed: {
+    isRequestTimerRunning() {
+      return this.timers.request.isRunning;
+    },
+  },
+
+  watch: {
+    isLocked(newValue) {
+      if (!newValue) {
+        this.stopRequestTimer();
+        return;
+      }
+
+      this.startRequestTimer();
+    },
+  },
+
+  methods: {
+    startRequestTimer() {
+      const initialCounter = this.timeout / 1000;
+
+      this.$emit('update:is-locked', true);
+      this.$emit('update:counter', initialCounter);
+      this.$timer.start('request');
+    },
+
+    stopRequestTimer() {
+      if (this.isRequestTimerRunning) {
+        this.$emit('update:is-locked', false);
+        this.$timer.stop('request');
+      }
+    },
+
+    onRequestTick() {
+      if (this.counter === 0) {
+        this.stopRequestTimer();
```

Comment:
also need `return` call here for not decrease value of counter

Reply:
Did not equal as you suggest. My variant fix annoying case when counter is `0`, but label still not hides. Now label hides after `1`, it's natural and logical (when counter is `0`, label should be already hided).